api/pkg/backend/service.go

```diff
@@ -447,3 +446,133 @@ func (s *APIService) listStorageType(ctx context.Context, request *restful.Reque
+	tier := &backend.Tier{}
+	err := request.ReadEntity(&tier)
+	if err != nil {
+		log.Errorf("failed to read request body: %v\n", err)
+		response.WriteError(http.StatusInternalServerError, err)
+		return
+	}
+	ctx := common.InitCtxWithAuthInfo(request)
+	actx := request.Attribute(c.KContext).(*c.Context)
+	tier.TenantId = actx.TenantId
+	tier.UserId = actx.UserId
+
+	res, err := s.backendClient.CreateTier(ctx, &backend.CreateTierRequest{Tier: tier})
+	if err != nil {
+		log.Errorf("failed to create tier: %v\n", err)
+		response.WriteError(http.StatusInternalServerError, err)
+		return
+	}
+
+	log.Info("Created backend successfully.")
+	response.WriteEntity(res.Tier)
+
+}
+
+//here backendId can be updated
+func (s *APIService) UpdateTier(request *restful.Request, response *restful.Response) {
+	log.Infof("Received request for updating tier: %v\n", request.PathParameter("id"))
+	id := request.PathParameter("id")
+	updateTierRequest := backend.UpdateTierRequest{Id: id}
+	err := request.ReadEntity(&updateTierRequest)
+	if err != nil {
+		log.Errorf("failed to read request body: %v\n", err)
+		response.WriteError(http.StatusInternalServerError, err)
+		return
+	}
+
+	ctx := common.InitCtxWithAuthInfo(request)
+	res, err := s.backendClient.UpdateTier(ctx, &updateTierRequest)
+	if err != nil {
+		log.Errorf("failed to update tier: %v\n", err)
+		response.WriteError(http.StatusInternalServerError, err)
+		return
+	}
+
+	log.Info("Update tier successfully.")
+	response.WriteEntity(res)
+}
+
+// GetTier if tierId is given then details of tier to be given
+func (s *APIService) GetTier(request *restful.Request, response *restful.Response) {
+	log.Infof("Received request for tier details: %s\n", request.PathParameter("id"))
+	id := request.PathParameter("id")
+
+	ctx := common.InitCtxWithAuthInfo(request)
+	res, err := s.backendClient.GetTier(ctx, &backend.GetTierRequest{Id: id})
+	if err != nil {
+		log.Errorf("failed to get tier details: %v\n", err)
+		response.WriteError(http.StatusInternalServerError, err)
+		return
+	}
+
+	log.Info("Get tier details successfully.")
+	response.WriteEntity(res.Tier)
+}
+
+//List of tiers is displayed
+func (s *APIService) ListTiers(request *restful.Request, response *restful.Response) {
+	log.Info("Received request for tier list.")
+
+	ctx := common.InitCtxWithAuthInfo(request)
+
+	listTierRequest := &backend.ListTierRequest{}
+
+	limit, offset, err := common.GetPaginationParam(request)
+	if err != nil {
+		log.Errorf("get pagination parameters failed: %v\n", err)
+		response.WriteError(http.StatusInternalServerError, err)
+		return
+	}
+	listTierRequest.Limit = limit
+	listTierRequest.Offset = offset
+
+	sortKeys, sortDirs, err := common.GetSortParam(request)
+	if err != nil {
+		log.Errorf("get sort parameters failed: %v\n", err)
+		response.WriteError(http.StatusInternalServerError, err)
+		return
+	}
+	listTierRequest.SortKeys = sortKeys
+	listTierRequest.SortDirs = sortDirs
+	log.Debug("***before list tier****")
```

Comment:
Remove all debug messages, if any there in other place also.

Reply:
done